### PR TITLE
Making the limit of records to retrieve per query in a batch query co…

### PR DESCRIPTION
**Pull request by hajoseng**
…nfigurable. This is because or Solr-Server has set the limit for "ORed" queries to a lower number (50) and we are forced to decrease the amount of records per batch query from 100 to 50.

### CODE REVIEW

File: config/vufind/config.ini
Comment (demiankatz) [inline config/vufind/config.ini]: Can you add `= 100` to the example here to make it more clear?

File: module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php]: If record_batch_size is not set in config.ini, this code will trigger a notice about an undefined value. I would suggest either wrapping the setPageSize() call in an `if (!empty(...)) {` check, or else providing a default value in the set call (i.e. `$config->Index->record_batch_size ?? 100`).

File: module/VuFindSearch/src/VuFindSearch/Backend/Solr/Backend.php
Comment (demiankatz) [inline module/VuFindSearch/src/VuFindSearch/Backend/Solr/Backend.php]: If you move the validation to the factory, you may not need this... but if you do want to validate, perhaps a better check would be whether the value is > 0, to avoid weird behavior caused by invalid configurations.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, this makes sense to me... just a few minor suggestions!

Also, it would be nice to add some test coverage for this new functionality (see https://github.com/vufind-org/vufind/blob/dev/module/VuFindSearch/tests/unit-tests/src/VuFindTest/Backend/Solr/BackendTest.php) -- I can help with that if you're not already familiar with the test suite. If you can give me write access to the branch, I can add a test directly; otherwise, I can update the test after merging the rest of the PR.

**Comment by demiankatz** [comment]
@hajoseng, one other thought: we already have a setting for limiting maximum boolean clauses:

https://github.com/vufind-org/vufind/blob/dev/config/vufind/config.ini#L550

Maybe the factory should take this into account, and if record_batch_size is larger than maxBooleanClauses, it should automatically adjust it. This would, of course, argue toward accounting for the default value of 100 in the factory.

File: module/VuFindSearch/src/VuFindSearch/Backend/Solr/Backend.php
Comment (hajoseng) [reply module/VuFindSearch/src/VuFindSearch/Backend/Solr/Backend.php]: Thank you Demian
I'm a little bit familiar with the test suite, but in this case I have no idea which test to add. I've given you write access to our repository to show me.

**Comment by demiankatz** [comment]
Thanks, @hajoseng -- I made a couple of small adjustments to the factory to prevent possible problems (see 61196db919bc0b7c4405c0eb74fe2cfaada0fa37) -- please review. I'll take care of the testing in the near future, as time permits.

**Review by demiankatz — APPROVED** [review]

**Comment by demiankatz** [comment]
@hajoseng, I've just added the test as promised. Please let me know if you have any questions/concerns. If you're happy with everything, I'll merge this as soon as I hear from you!

**Comment by hajoseng** [comment]
@demiankatz, thank you very much; I'm happy with your adjustments.

**Comment by demiankatz** [comment]
Thanks; merging now!

**Pull request merged**
